Lib9c/Action/SellCancellation.cs

```diff
@@ -134,9 +134,13 @@ public override IAccountStateDelta Execute(IActionContext context)
             Order order = OrderFactory.Deserialize(orderDict);
             order.ValidateCancelOrder(avatarState, tradableId);
             ITradableItem sellItem = order.Cancel(avatarState, context.BlockIndex);
-            var shardedShopState = new ShardedShopStateV2(shopStateDict);
-            shardedShopState.Remove(order, context.BlockIndex);
-            states = states.SetState(shardedShopAddress, shardedShopState.Serialize());
+            if (sellItem.RequiredBlockIndex < context.BlockIndex)
```

Comment:
it would be better use contains instead of blockIndex

Reply:
`ITradableItem sellItem = order.Cancel(avatarState, context.BlockIndex);`
When canceling the order, the `RequiredBlockIndex` of sellItem is changed to `context.BlockIndex`.